frontend/src/views/material/MaterialOverview.vue

```diff
@@ -0,0 +1,168 @@
+              <template #activator="{ attrs, on }">
+                <v-list-item v-bind="attrs" v-on="on">
+                  <v-list-item-icon>
+                    <v-icon>mdi-plus</v-icon>
+                  </v-list-item-icon>
+                  <v-list-item-content
+                    >{{ $tc('views.material.materialOverview.createNewList') }}
+                  </v-list-item-content>
+                </v-list-item>
+              </template>
+            </DialogMaterialListCreate>
+            <v-list-item @click="downloadXlsx">
+              <v-list-item-icon>
+                <v-icon>mdi-microsoft-excel</v-icon>
+              </v-list-item-icon>
+              <v-list-item-content
+                >{{ $tc('views.material.materialOverview.download') }}
+              </v-list-item-content>
+            </v-list-item>
+          </v-list>
+        </v-menu>
+      </template>
+      <v-expansion-panels
+        v-if="camp().periods().items.length > 1"
+        v-model="openPeriods"
+        multiple
+        flat
+        accordion
+      >
+        <PeriodMaterialLists
+          v-for="period in camp().periods().items"
+          :key="period._meta.self"
+          :period="period"
+          :material-item-collection="period.materialItems()"
+        />
+      </v-expansion-panels>
+      <v-card-text v-else-if="camp().periods().items.length === 1">
+        <MaterialTable
+          :camp="camp()"
+          :material-item-collection="camp().periods().items[0].materialItems()"
+          :period="camp().periods().items[0]"
+          :disabled="!isContributor"
+        />
+      </v-card-text>
+    </content-card>
+  </v-container>
+</template>
+
+<script>
+import ContentCard from '@/components/layout/ContentCard.vue'
+import MaterialTable from '@/components/material/MaterialTable.vue'
+import * as XLSX from 'xlsx'
+import PeriodMaterialLists from '@/components/material/PeriodMaterialLists.vue'
+import { campRoleMixin } from '@/mixins/campRoleMixin.js'
+import DialogMaterialListCreate from '@/components/campAdmin/DialogMaterialListCreate.vue'
+import slugify from 'slugify'
+
+export default {
+  name: 'MaterialOverview',
+  components: {
+    DialogMaterialListCreate,
+    PeriodMaterialLists,
+    MaterialTable,
+    ContentCard,
+  },
+  mixins: [campRoleMixin],
+  props: {
+    camp: { type: Function, required: true },
+  },
+  data() {
+    return {
+      openPeriods: [],
+    }
+  },
+  computed: {
+    collection() {
+      return this.api.get().materialItems({
+        camp: this.camp()._meta.self,
+      })
+    },
+  },
+  mounted() {
+    this.camp().activities().$reload()
+    this.camp()
+      .periods()
+      ._meta.load.then((periods) => {
+        this.openPeriods = periods.items.reduce((result, period, index) => {
+          if (Date.parse(period.end) >= new Date()) {
+            result.push(index)
+          }
+          return result
+        }, [])
+      })
+  },
+  methods: {
+    async downloadXlsx() {
```

Comment:
This large block of logic seems duplicated. Could you move this to a mixin or something similar?

Reply:
[34acd22](https://github.com/ecamp/ecamp3/pull/3797/commits/34acd225db8c66ed906743124b599de6e1f4fbbf)